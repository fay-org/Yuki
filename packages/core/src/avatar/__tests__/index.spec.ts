import { mount } from '@vue/test-utils'
import Avatar from '..'

const BASE_IMG_URL = 'http://img.com'

describe('Avatar', () => {
  it('render normal', () => {
    const wrapepr = mount(Avatar)
    expect(() => wrapepr.unmount()).not.toThrow()
  })
  it('img', () => {
    const wrapper = mount(Avatar, {
      props: {
        src: BASE_IMG_URL
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('text', () => {
    const wrapper = mount(Avatar, {
      props: {
        text: 'kanno'
      }
    })
    expect(wrapper.find('span').text()).toBe('kan')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('safe text', () => {
    const wrapper = mount(Avatar, {
      props: {
        text: 123 as unknown as string
      }
    })
    expect(wrapper.find('span').text()).toBe('')
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('scale', () => {
    const wrapper = mount(Avatar, {
      props: {
        h: 3,
        text: 'S'
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('square', () => {
    const wrapper = mount(Avatar, {
      props: {
        isSquare: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('stacked', () => {
    const wrapper = mount(Avatar, {
      props: {
        stacked: true
      }
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})
