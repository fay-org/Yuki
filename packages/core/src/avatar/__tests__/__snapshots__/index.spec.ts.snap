// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Avatar img 1`] = `"<div class="fect-avatar" style="--avatar-border-color: #eaeaea; --avatar-bg-color: #fff; --avatar-radius: 6px; --avatar-size: calc(1.75 * 16px); --avatar-pt: 0; --avatar-pr: 0; --avatar-pb: 0; --avatar-pl: 0; --avatar-mt: 0; --avatar-mr: 0; --avatar-mb: 0; --avatar-ml: 0; --avatar-font-size: calc(1 * 16px);"><img src="http://img.com" draggable="false"></div>"`;

exports[`Avatar safe text 1`] = `"<div class="fect-avatar" style="--avatar-border-color: #eaeaea; --avatar-bg-color: #fff; --avatar-radius: 6px; --avatar-size: calc(1.75 * 16px); --avatar-pt: 0; --avatar-pr: 0; --avatar-pb: 0; --avatar-pl: 0; --avatar-mt: 0; --avatar-mr: 0; --avatar-mb: 0; --avatar-ml: 0; --avatar-font-size: calc(1 * 16px);"><span class="fect-avatar__text"></span></div>"`;

exports[`Avatar scale 1`] = `"<div class="fect-avatar" style="--avatar-border-color: #eaeaea; --avatar-bg-color: #fff; --avatar-radius: 6px; --avatar-size: calc(1.75 * 16px); --avatar-pt: 0; --avatar-pr: 0; --avatar-pb: 0; --avatar-pl: 0; --avatar-mt: 0; --avatar-mr: 0; --avatar-mb: 0; --avatar-ml: 0; --avatar-font-size: calc(1 * 16px);"><span class="fect-avatar__text">S</span></div>"`;

exports[`Avatar square 1`] = `"<div class="fect-avatar fect-avatar--square" style="--avatar-border-color: #eaeaea; --avatar-bg-color: #fff; --avatar-radius: 6px; --avatar-size: calc(1.75 * 16px); --avatar-pt: 0; --avatar-pr: 0; --avatar-pb: 0; --avatar-pl: 0; --avatar-mt: 0; --avatar-mr: 0; --avatar-mb: 0; --avatar-ml: 0; --avatar-font-size: calc(1 * 16px);"><span class="fect-avatar__text"></span></div>"`;

exports[`Avatar stacked 1`] = `"<div class="fect-avatar fect-avatar--stacked" style="--avatar-border-color: #eaeaea; --avatar-bg-color: #fff; --avatar-radius: 6px; --avatar-size: calc(1.75 * 16px); --avatar-pt: 0; --avatar-pr: 0; --avatar-pb: 0; --avatar-pl: 0; --avatar-mt: 0; --avatar-mr: 0; --avatar-mb: 0; --avatar-ml: calc(-0.625 * 16px); --avatar-font-size: calc(1 * 16px);"><span class="fect-avatar__text"></span></div>"`;

exports[`Avatar text 1`] = `"<div class="fect-avatar" style="--avatar-border-color: #eaeaea; --avatar-bg-color: #fff; --avatar-radius: 6px; --avatar-size: calc(1.75 * 16px); --avatar-pt: 0; --avatar-pr: 0; --avatar-pb: 0; --avatar-pl: 0; --avatar-mt: 0; --avatar-mr: 0; --avatar-mb: 0; --avatar-ml: 0; --avatar-font-size: calc(1 * 16px);"><span class="fect-avatar__text">kan</span></div>"`;
